backend/question-service/src/questions/questions.model.ts

```diff
@@ -0,0 +1,84 @@
+import { BadRequestException, Injectable, InternalServerErrorException, NotFoundException } from '@nestjs/common';
+import { CreateQuestionDto } from './dto/create-question.dto';
+import { UpdateQuestionDto } from './dto/update-question.dto';
+import { Question, QuestionDocument } from './schemas/question.schema';
+import { InjectModel } from '@nestjs/mongoose';
+import { FilterQuery, Model, Types } from 'mongoose';
+import { FilterQuestionDto } from './dto/filter-question.dto';
+
+@Injectable()
+export class QuestionDB {
+  constructor(
+    @InjectModel(Question.name) private questionModel: Model<QuestionDocument>,
+  ) {}
+
+  async createQuestionInDB(
+    createQuestionDto: CreateQuestionDto,
+  ): Promise<Question> {
+    const newQuestion = new this.questionModel(createQuestionDto);
+    return newQuestion.save();
+  }
+
+  async findAllQuestionsInDB(filterDto?: FilterQuestionDto): Promise<Partial<Question>[]> {
+    const filterQuery: FilterQuery<QuestionDocument> = {};
+  
+    if (filterDto?.categories && filterDto.categories.length > 0) {
+      filterQuery.categories = { $all: filterDto.categories };
+    }
+  
+    if (filterDto?.complexity) {
+      filterQuery.complexity = filterDto.complexity;
+    }
+  
+    const query = this.questionModel.find(filterQuery);
+  
+    return await query.select('questionId title categories complexity').exec();
```

Comment:
I think questionId field is no longer there in the schema, perhaps have to change it to _id